### PR TITLE
string changed to uuuid and exception handling added

### PR DESCRIPTION
**Pull request by Adam3004**

### CODE REVIEW

File: src/main/java/com/brightpath/learnify/domain/mapper/DtoMapper.java [hunk 1]
```diff
@@ -1,4 +1,4 @@
-package com.brightpath.learnify.controller;
+package com.brightpath.learnify.domain.mapper;
```
Comment (krisx303) [inline src/main/java/com/brightpath/learnify/domain/mapper/DtoMapper.java]: Why was this file moved? I dont think this mapper should be part of the domain layer

File: src/main/java/com/brightpath/learnify/controller/QuizController.java [hunk 2]
```diff
@@ -36,43 +37,43 @@ public class QuizController implements QuizzesApi {
...
 
     @Override
-    public ResponseEntity<QuizSummaryDto> showQuizById(String quizId) {
-        Optional<Quiz> quiz = quizService.showQuizById((UUID.fromString(quizId)));
-        return quiz
-                .map(quizToConvert -> ResponseEntity.status(OK).body(quizToConvert.convertToQuizSummaryDto()))
-                .orElseGet(() -> ResponseEntity.notFound().build());
+    public ResponseEntity<QuizSummaryDto> showQuizById(UUID quizId) {
+        Quiz quiz = quizService.showQuizById(quizId);
+        return ResponseEntity
```
Comment (krisx303) [inline src/main/java/com/brightpath/learnify/controller/QuizController.java:58]: nitpick: Can we use ResponseEntity.ok() instead of the builder-like style to maintain consistency?

File: src/main/java/com/brightpath/learnify/exception/notfound/ResourceType.java [hunk 3]
```diff
@@ -0,0 +1,17 @@
+package com.brightpath.learnify.exception.notfound;
+
+import lombok.Getter;
+
+@Getter
+public enum ResourceType {
+    NOTE("note"),
+    QUIZ("quiz"),
+    NOTE_PAGE("note page"),
```
Comment (krisx303) [inline src/main/java/com/brightpath/learnify/exception/notfound/ResourceType.java]: Maybe `BOARD_PAGE` instad of `NOTE_PAGE` or even `BOARD_NOTE_PAGE` and `DOCUMENT_NOTE_PAGE`?

File: src/main/resources/api.yaml [hunk 4]
```diff
@@ -58,7 +58,7 @@ paths:
             application/json:
               schema:
                 $ref: '#/components/schemas/NoteSummaryDto'
-  /notes/recent:
+  /notes/recent: #todo we should remeber to add userId when it will exists
```
Comment (krisx303) [inline src/main/resources/api.yaml:61]: I'm not sure about this. We could do it this way, but are we planning to allow users to view others recently viewed notes? For other endpoints we'll need to check if the user has permission to access the resource, so we could also extract userId from the token here. WDYT?

File: src/main/java/com/brightpath/learnify/domain/quiz/question/Question.java [hunk 5]
```diff
@@ -46,7 +46,7 @@ public Question(QuestionDto questionDto, UUID quizId, UUID questionId) {
     }
 
     public QuestionDto convertToQuestionDto() {
```
Comment (krisx303) [inline src/main/java/com/brightpath/learnify/domain/quiz/question/Question.java]: I think we should move that to DtoMapper

File: src/main/java/com/brightpath/learnify/exception/mapper/ExceptionMapper.java [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+
+@ControllerAdvice
+public class ExceptionMapper {
+    private static final String BAD_REQUEST_DEFAULT_MESSAGE = "Bad request, reason: ";
+
+    //unauthorized
+    @ExceptionHandler
+    public ResponseEntity<?> onAuthenticationToGetFailedException(UserNotAuthorizedToGetException ex) {
+        return ResponseEntity
+                .status(UNAUTHORIZED)
```
Comment (krisx303) [inline src/main/java/com/brightpath/learnify/exception/mapper/ExceptionMapper.java]: I know this is nitpicking but if user is not authorized to access resource but is authenticated then response code should be 403 FORBIDDEN instead of 401 UNAUTHORIZED

File: src/main/java/com/brightpath/learnify/domain/binding/BindingService.java [hunk 7]
```diff
@@ -42,10 +44,15 @@ public List<Note> listNotesBoundToQuiz(UUID quizId) {
...
 
     public List<Quiz> listQuizzesBoundToNote(UUID noteId) {
+        checkIfNoteExists(noteId);
         List<QuizEntity> bindings = bindingRepository.findAllBoundQuizzesByNoteId(noteId);
         return bindings.stream()
                 .map(persistentMapper::asQuiz)
                 .toList();
     }
 
+    private void checkIfNoteExists(UUID noteId) {
```
Comment (krisx303) [inline src/main/java/com/brightpath/learnify/domain/binding/BindingService.java:57]: We probably need a similar check for quizzes when requesting bound notes

File: src/main/java/com/brightpath/learnify/domain/quiz/QuizService.java [hunk 8]
```diff
@@ -62,9 +66,12 @@ public List<Quiz> listRecentQuizzes() {
                 .toList();
     }
 
-    public Optional<QuizSimpleResult> updateQuizResult(UUID quizId, QuizSimpleResult quizSimpleResult) {
+    public QuizSimpleResult updateQuizResult(UUID quizId, QuizSimpleResult quizSimpleResult) {
+        if (findQuizEntity(quizId).isEmpty()) {
+            throw new ResourceNotFoundException(QUIZ);
+        }
         if (quizSimpleResult == null) {
```
Comment (krisx303) [inline src/main/java/com/brightpath/learnify/domain/quiz/QuizService.java:73]: Is there any case when `quizSimpleResult` is null?

**Review by krisx303 — COMMENTED** [review]
Mostly nitpicking from my side, but the code looks good. My main doubt is whether we should change the status code from 401 to 403 ?

File: src/main/java/com/brightpath/learnify/domain/mapper/DtoMapper.java (hunk 1 above)
Comment (Adam3004) [reply src/main/java/com/brightpath/learnify/domain/mapper/DtoMapper.java]: Imo didn't match to the controller package, maybe we can find some better place for it?

File: src/main/java/com/brightpath/learnify/exception/mapper/ExceptionMapper.java (hunk 6 above)
Comment (Adam3004) [reply src/main/java/com/brightpath/learnify/exception/mapper/ExceptionMapper.java]: right

File: src/main/java/com/brightpath/learnify/domain/quiz/QuizService.java (hunk 8 above)
Comment (Adam3004) [reply src/main/java/com/brightpath/learnify/domain/quiz/QuizService.java:73]: yes, when quiz was just created and there is not any results yet

File: src/main/java/com/brightpath/learnify/controller/QuizController.java (hunk 2 above)
Comment (Adam3004) [reply src/main/java/com/brightpath/learnify/controller/QuizController.java:58]: it is like this, because ReponseEntity.created() requires URI as a parametr so i maintained consistency for this class. Is that ok or?

File: src/main/resources/api.yaml (hunk 4 above)
Comment (Adam3004) [reply src/main/resources/api.yaml:61]: right, we should allow this

File: src/main/java/com/brightpath/learnify/exception/notfound/ResourceType.java [hunk 9]
```diff
@@ -6,8 +6,8 @@
 public enum ResourceType {
     NOTE("note"),
     QUIZ("quiz"),
-    NOTE_PAGE("note page"),
-    DOCUMENT_PAGE("document page");
+    BOARD_PAGE("note page"),
+    BOARD_NOTE_PAGE("document page");
```
Comment (krisx303) [inline src/main/java/com/brightpath/learnify/exception/notfound/ResourceType.java:10]: Should be BOARD_NOTE_PAGE and DOCUMENT_NOTE_PAGE

File: src/main/java/com/brightpath/learnify/domain/mapper/DtoMapper.java (hunk 1 above)
Comment (krisx303) [reply src/main/java/com/brightpath/learnify/domain/mapper/DtoMapper.java]: Mapping between the domain and API layers don’t belong in the domain. The domain should only handle business logic, not how data is sent or received (just like persistance layer and persistent mapper - those are seperated from domain package)

File: src/main/java/com/brightpath/learnify/controller/QuizController.java (hunk 2 above)
Comment (krisx303) [reply src/main/java/com/brightpath/learnify/controller/QuizController.java:58]: Using CREATED instead OK is another thing we should change (not in scope of this PR) but here I suggested to replace `ResponseEntity.status(OK).body(...)` with `ResponseEntity.ok(...)` to maintain consistency between controllers

File: src/main/java/com/brightpath/learnify/domain/binding/BindingService.java (hunk 7 above)
Comment (krisx303) [reply src/main/java/com/brightpath/learnify/domain/binding/BindingService.java:57]: What about this?

**Review by krisx303 — APPROVED** [review]
